src/query/v2/shard_request_manager.hpp

```diff
@@ -649,6 +699,38 @@ class ShardRequestManager : public ShardRequestManagerInterface {
     }
   }
 
+  void AwaitOnResponses(ExecutionState<GetPropertiesRequest> &state, std::vector<GetPropertiesResponse> &responses) {
```

Comment:
This is almost identical to `Expand`. However, I did not bother to simplify this as Tyler is working on simplifying the `ShardRequestManager`. I will create another PR once it's clear what will change/removed